src/file/FileController.js

```diff
@@ -264,6 +270,19 @@ export class FileController {
 			return zip.generateAsync({type: 'uint8array'})
 		}).then(zf => createDataFile(file, zf))
 	}
+
+	/**
+	 * Export a list of entities of type T. Export format will be decided by the type ref T._type
+	 * @param entities array of any type with a type ref
+	 */
+	exportEntities<T>(entities: T[]): void {
```

Comment:
```suggestion
	exportEntities<T: Element | ListElement>(entities: T[]): void {
```

Then you don't need to downcast before filtering.

Reply:
Or rather:

```suggestion
	exportEntities<T: {_type: TypeRef<T>}>(entities: T[]): void {
```

Reply:
I just noticed that we import fileController directly from the list. We should absolutely not do this, sory I didn't notice before, we won't be able to use List in some situations then. We should put dragStart on `ListConfig` as optional property and then only define it for Mail list. You don't need a typeRef check then and you can call it `exportMails()`.

If you *would* write a type for it then I would do:
```ts
export type TypeRefd<T> = T & {
	_type: TypeRef<T>,
}

//...

export class FileController {
  exportEntities<E, T: TypeRefd<E>>(entities: $ReadOnlyArray<T>): void {
    //...
  }
}


const mails: Array<Mail> = []
const bananas: Array<String> = []
fileController.exportEntities(mails)
// $FlowExpectedError
fileController.exportEntities(bananas)
```